source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+       ``BsonProperty`` annotation to link the parameters with properties.
+
+   * - ``BsonDiscriminator``
+     - Specifies a class uses a discriminator. You can set a custom
+       discriminator key and value.
+
+   * - ``BsonRepresentation``
+     - Specify a BSON type to store when different from the POJO property.
+
+   * - ``BsonId``
+     - Marks a property to serialized as the _id property.
+
+   * - ``BsonIgnore``
+     - Marks a property to ignore. You can configure whether to serialized
+       and/or deserialize a property.
+
+   * - ``BsonProperty``
+     - Allows for an alternative document key name when converting the POJO
+       field to BSON. Also allows a field to turn on using a discriminator
+       when storing a POJO value.
+
+The following code snippet shows a sample POJO called ``Product`` that uses
+the annotations described above.
+
+.. _bson-annotation-code-example:
+
+.. code-block:: java
+   :emphasize-lines: 9,11,14,15,18,21
+
+   import org.bson.BsonType;
+   import org.bson.codecs.pojo.annotations.BsonCreator;
+   import org.bson.codecs.pojo.annotations.BsonDiscriminator;
+   import org.bson.codecs.pojo.annotations.BsonId;
+   import org.bson.codecs.pojo.annotations.BsonIgnore;
+   import org.bson.codecs.pojo.annotations.BsonProperty;
+   import org.bson.codecs.pojo.annotations.BsonRepresentation;
+
+   @BsonDiscriminator(value="AnnotatedProduct", key="_cls")
+   public class Product {
+       @BsonProperty("modelName")
+       private String name;
+
+       @BsonId()
+       @BsonRepresentation(BsonType.OBJECT_ID)
+       private String serialNumber;
+
+       @BsonIgnore
+       private List<Product> relatedItems;
+
+       @BsonCreator
+       public ProductPojo(@BsonProperty("modelName") String name) {
+           this.name = name;
+       }
+
+       // ...
+
+   }
+
+.. _bson-annotation-convention:
+
+.. note::
 
-Advanced configuration
+   When using annotations, remember to specify the
+   ``Conventions.ANNOTATION_CONVENTION`` in your ``ClassModelBuilder``
+   or ``PojoCodecProvider.Builder``. For example:
+
+   .. code-block:: java
+
+      ClassModel<Product> classModel = ClassModel.builder(Product.class).
+             conventions(Arrays.asList(Conventions.ANNOTATION_CONVENTION)).build();
+
+The annotations in the example POJO specify the following behavior:
+
+- reference the POJO with the specified discriminator key and   value, adding
+  the "_cls" field with the value of "AnnotatedProduct" to the BSON document
+  on write operations
+- convert between the POJO ``name`` field and value and the BSON ``modelName``
+  field and value in the document
+- convert between the POJO ``serialNumber`` field and value the BSON document
+  ``_id`` field and value in the document
+- omit the ``relatedItems`` field and value when converting data
+- use the ``Product(String name)`` constructor when instantiating the POJO
+
+.. _pojo-discriminators:
+
+Discriminators
```

Comment:
Discriminators are mentioned an awful lot in the preceding sections. I would consider either moving this section above or explicitly linking to this section when you mention discriminators, otherwise reading this page in a linear format can make you think that you're missing prerequisite knowledge about discriminators. Or maybe a link/mention in the overview would be enough?

Reply:
Will add a sentence and link in the overview, thanks for the suggestion.